emcc.py

```diff
@@ -1368,6 +1368,10 @@ def check(input_file):
       shared.Settings.OPT_LEVEL = opt_level
       shared.Settings.DEBUG_LEVEL = debug_level
 
+      if not shared.Settings.LEGALIZE_JS_FFI:
```

Comment:
please move this block to around like 1366 (so it's before those 3 assigments it is after right now).